scripts/s3_unshard_to_hf.py

```diff
@@ -31,91 +31,126 @@ def make_parser():
+    )
+    parser.add_argument(
+        "--old_style_hf", action="store_true", help="If given, convert to 'old-style' HF checkpoint."
+    )
     parser.add_argument(
         "--quiet",
         action="store_true",
         help="If given, don't show progress for AWS commands.",
     )
     parser.add_argument("--type", default=None, help="If given, pass this argument on to `unshard.py`.")
-    parser.add_argument("--model-only", action="store_true", help="If given, only unshard the model.")
+    parser.add_argument("--model_only", action="store_true", help="If given, only unshard the model.")
     return parser
 
 
-def aws_copy(src, dest, quiet):
-    base = "aws s3 cp --recursive"
-    if quiet:
+def aws_copy(src, dest, args):
+    base = "aws s3 sync --exclude tmp/*"
+    if args.quiet:
         base += " --quiet"
+    if args.type == "olmo_core" and args.model_only:
+        # Don't copy optimizer and trainer state if we're only unsharding the model.
+        base += " --exclude optim/* --exclude train/*"
     cmd = f"{base} {src} {dest}"
 
     return cmd
 
 
 def s3_unshard_to_hf(args):
     # Set directories
-    sharded_dir = args.tmp_dir / "sharded"
-    unsharded_dir = args.tmp_dir / "unsharded"
-    hf_dir = args.tmp_dir / "hf"
-    hf_dir.mkdir()
-
-    # Download sharded checkpoint.
-    print("Downloading sharded checkpoint from S3.")
-    download_cmd = aws_copy(args.sharded_bucket, sharded_dir, args.quiet)
-    subprocess.run(download_cmd, shell=True, check=True)
-
-    # Unshard.
-    print("Unsharding.")
-    unshard_cmd = f"python scripts/unshard.py {sharded_dir} {unsharded_dir}"
-    # Add a `--type` argument if given.
-    if args.type is not None:
-        unshard_cmd += f" --type {args.type}"
-    if args.model_only:
-        unshard_cmd += " --model-only"
-
-    subprocess.run(unshard_cmd, shell=True, check=True)
-
-    # Convert to HF
+    sharded_dir = args.local_dir / "sharded"
+    unsharded_dir = args.local_dir / "unsharded"
+    hf_dir = args.local_dir / "hf"
+    hf_dir.mkdir(exist_ok=True)
+
+    # Either download the unsharded checkpoint, or download sharded and unshard.
+    if args.already_unsharded:
+        download_cmd = aws_copy(args.unsharded_bucket, unsharded_dir, args)
+        subprocess.run(download_cmd, shell=True, check=True)
+    else:
+        if not args.already_downloaded:
+            # Download sharded checkpoint.
+            print("Downloading sharded checkpoint from S3.")
+            download_cmd = aws_copy(args.sharded_bucket, sharded_dir, args)
+            subprocess.run(download_cmd, shell=True, check=True)
+
+        # Unshard.
+        print("Unsharding.")
+        unshard_cmd = f"python scripts/unshard.py {sharded_dir} {unsharded_dir}"
+        # Add a `--type` argument if given.
+        if args.type is not None:
+            unshard_cmd += f" --type {args.type}"
+        if args.model_only:
+            unshard_cmd += " --model-only"
```

Comment:
nit: `model_only`

Reply:
I was matching the hyphen in [unshard.py](https://github.com/allenai/OLMo/blob/main/scripts/unshard.py#L90), is this incorrect practice?

Reply:
My bad, I saw you change `model-only` to `model_only` a few lines earlier so I commented without thinking much about it